envs/env.py

```diff
@@ -336,6 +346,19 @@ def _start_external_processes(self, reset_time=True):
             self.episode_start_time = time.time()
         logging.info("All commands executed. Episode started!")
 
+    def set_random_variation_step(self):
+        random_step = int(self.max_time_steps_per_episode/3) if self._is_testing else np.random.randint(self.max_time_steps_per_episode/4, 3*self.max_time_steps_per_episode/4)
```

Comment:
See comment on line 120. 
Also, you can use that in `__init__` as well if you need it instead of duplicating code.